scripts/lsstinstall

```diff
@@ -0,0 +1,348 @@
+#!/bin/sh
```

Comment:
The intend is to use sh compared to newinstall.sh that used bash for POSIX compliance.
Ubuntu 20.04 LTS links /bin/sh to /bin/dash and lsstinstall fails with
`../lsstinstall: 5: /home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/etc/conda/activate.d/activate-binutils_linux-64.sh: Syntax error: "(" unexpected
` 
Invoking bash on Ubuntu 20.04 LTS works. MacOS arm works as well.
CentOS7 links /bin/sh to bash

Reply:
Hm. That seems to be in conda's activation, which is not really my fault.  Let me look into it, but it may need to be fixed on their side.

Reply:
The problem seems to be activate-binutils_linux-64.sh
requiring bash. By default the executable flag is not set and this seems to be sourced from the current shell session and fail.

Reply:
So for MacOS /bin/sh is actually zsh.
`/bin/sh
echo $SHELL
zsh`
Similar for Centos, Ubuntu. What you request is  not what you get.
There seems to be no clean way to get a real sh. Requiring bash seems to be to lowest common denominator.

Reply:
I was under the impression that when bash runs as sh it falls back to sh compatibility. Is that not what happens? And what do you get if you run with a blank line instead of shebang (which always used to invoke the posix compatible shell I thought).

Reply:
I don't know how this ever works for `conda activate` in a non-bash shell, then.  This activation script (and other similar ones for conda compilers) only have bash versions and are definitely bash-specific and have been for 2 years.

The problem is not /bin/sh being other shells; that I'm expecting and so used the POSIX lowest-common-denominator in the lsstinstall code.  We could insist that lsstinstall be run using bash, but it shouldn't need to be.

Reply:
Well,  activate-binutils_linux-64.sh is not POSIX compliant (bash) and lsstinstall fails when using dash on Ubuntu, which aims to be POSIX compliant, so any invocation of sh on MacOS/CentOS (bash. zsh) allows to source the bash script, so strictly speaking the  "fallback" solution is broken in allowing this.

Reply:
Looks like the activation functions would be POSIX- and dash-compatible if they just left out the `function` keyword.

Reply:
Aha!

```
# ./conda/bin/conda init dash

ArgumentError: Invalid shells:
  - dash

Currently available shells are:
  - bash
  - fish
  - powershell
  - tcsh
  - xonsh
  - zsh
```
So now I'm blaming:
```
./conda/bin/conda shell.dash hook
```
for not failing.

Reply:
Also looks like I may not have taken arm64/aarch64 into account properly.

Reply:
https://github.com/conda-forge/ctng-compiler-activation-feedstock/pull/59 should fix this.

Reply:
I think the only thing holding this up is non-strict-POSIX-ness in the eups activate script.  Is that correct, @mwittgen? After tracing through the logic, I think I'm OK with the ARM architectures; they should fall through to eups "flavor" determination which should in turn fall back to source since no binaries are available.

Reply:
Correct,  @ktlim.